docs/patterns/ingress.md

```diff
@@ -167,8 +169,78 @@ spec:
           serviceName: bookstore-v1
           servicePort: 8080 # Note: port 80 cannot be used for HTTPS ingress with Azure Application Gateway ingress
 ```
+### Using Gloo API Gateway
+
+[Gloo API Gateway][5] is an Envoy-powered API gateway that can run in `Ingress` mode or full-blow `Gateway` mode. In this document, we show the `Ingress` approach, but you can refer to the [Gloo documentation][5] for more in depth functionality enabled by Gloo. 
+
+Install Gloo in `Ingress` mode:
+
+```bash
+glooctl install ingress
+```
+
+For Gloo's ingress, we don't need any additional annotations, however, the `kubernetes.io/ingress.class: gloo` annotation is recommended. With Gloo, we configure the `Upstream` objects with the appropriate trust authority. In Gloo, the `Upstream` object represents a routable target (Kubernetes Service, Consul Service, Cloud Function, etc). 
+
+To prepare the root certifciate, we must do something similar to what we do for the Azure Application Gateway.
+
+```bash
+kubectl get secret/osm-ca-bundle -n osm-system -o jsonpath="{.data['ca\.crt']}" | base64 -d > osm-c-bundlea.pem
+
+glooctl create secret tls --name osm-ca-bundle --rootca osm-c-bundlea.pem 
```

Comment:
OSM stores the root certificate in a secret `osm-cabundle` in the `osm-system` namespace by default. Is it possible to reference this secret directly with Gloo based ingress? This would be similar to how `nginx-ingress` is configured with OSM.

Reply:
Yah Gloo doesn't seem to recognize that secret because it's not a 'tls' secret... let me check the code real quick to double check

Reply:
Sounds good. Were you able to test ingess with OSM and Gloo?

Reply:
Yah, gloo expects secrets a certain way, so no updates on OSM side. We will work internally to make this a little easier though and send updated PR at a later date.